src/app/modules/courses/components/course-add-edit/course-add-edit.component.html

```diff
@@ -10,6 +10,10 @@ <h2>{{currentCourseId === null ? 'New course' : (form.value['name'] | uppercase)
       type="text"
       placeholder="Text input"
     />
+    <span *ngIf="!!controls['name'].hasError('required')" class="error">Name is required!</span>
```

Comment:
Maybe you can create component, which will render list of errors and needed text based on error type?